include/settings.hpp

```diff
@@ -0,0 +1,428 @@
+#ifndef GRAPH_PROTOTYPE_SETTINGS_HPP
+#define GRAPH_PROTOTYPE_SETTINGS_HPP
+
+#include <atomic>
+#include <chrono>
+#include <concepts>
+#include <mutex>
+#include <optional>
+#include <reflection.hpp>
+#include <set>
+#include <tag.hpp>
+#include <variant>
+
+namespace fair::graph {
+
+struct SettingsCtx {
+    // using TimePoint = std::chrono::time_point<std::chrono::utc_clock>; // TODO: change once the C++20 support is ubiquitous
+    using TimePoint               = std::chrono::time_point<std::chrono::system_clock>;
+    std::optional<TimePoint> time = std::nullopt; /// UTC time-stamp from which the setting is valid
+    std::string              context;             /// user-defined multiplexing context for which the setting is valid
```

Comment:
I think we had a discussion yesterday about changing this to `tag_t`, just adding this here as a note, since the basic_setting implementation does not use it this can be revisited when we tackle the context aware setting implementation.